blueocean-dashboard/src/main/js/components/testing/TestLogService.js

```diff
@@ -0,0 +1,55 @@
+import { BunkerService, Fetch } from '@jenkins-cd/blueocean-core-js';
+
+class Item {
+    constructor(test, log, isStdOut) {
+        this.test = test;
+        this.log = log;
+        this.isStdOut = isStdOut;
+    }
+}
+
+export default class TestLogService extends BunkerService {
+
+    loadStdOut(test) {
+        if (this.hasItem(test)) return;
+        Fetch.fetch(test._links.stdout.href)
+            .then(res => res.text())
+            .then((data) => {
+                this.setItem(new Item(test, data, true));
+            })
+            .catch(TestLogService.ignoreNotFound);
```

Comment:
Use of `catch` like this can swallow errors caused by any prior `then` call.  If `this.setItem` triggers a React render and an exception occurs (either in `observer` component or one of its descendants) then this catch block will fire, even though it's really intended just to handle a 404. If we are just concerned about handling a 404 from the REST API, the `ignoreNotFound` method should be passed as the second argument to the first `then` call.